### ISSUE TITLE
Cannot start desktop environment for ubuntu Jammy

### ISSUE THREAD

**Issue by hungshe** [dnschneid/crouton#4908]
```
Please paste the output of the following command here: sudo edit-chroot -all
```
xxx@localhost: sudo edit-chroot -l jammy
...
WARNING: Unable to register core dump handler.
crouton: version 1-20221126232257~master:ae594340
release: jammy
architecture: arm64
xmethod: xorg
targets: audio,cli-extra,core,xfce-desktop,x11,xorg
host: version 15183.69.0 (Official Build) stable-channel strongbad 
kernel: Linux localhost 5.15.73-12721-g62740cbc4ac1 #1 SMP PREEMPT Wed Dec 7 23:29:18 PST 2022 aarch64 aarch64 aarch64 GNU/Linux
freon: yes

#### Please describe your issue:

/usr/bin/startxfce4: Starting X server

_XSERVTransmkdir: Owner of /tmp/.X11-unix should be set to root

X.Org X Server 1.21.1.3
X Protocol Version 11, Revision 0
Current Operating System: Linux localhost 5.15.73-12721-g62740cbc4ac1 #1 SMP PREEMPT Wed Dec 7 23:29:18 PST 2022 aarch64
Kernel command line: cros_secure console= loglevel=7 init=/sbin/init cros_secure drm.trace=0x106 root=/dev/dm-0 rootwait ro dm_verity.error_behavior=3 dm_verity.max_bios=-1 dm_verity.dev_wait=1 dm="1 vroot none ro 1,0 4710400 verity payload=PARTUUID=f935fdb6-1adb-4cf0-a184-1ec91e16c2dd/PARTNROFF=1 hashtree=PARTUUID=f935fdb6-1adb-4cf0-a184-1ec91e16c2dd/PARTNROFF=1 hashstart=4710400 alg=sha256 root_hexdigest=d0ee25ef2b0e3d2c416c97f79f0d43623162162f9cf9ffc4a9254e5ececb1a8e salt=5e3fe920f8f57c47c3fc4823f0e924ccc0326cd3234728d25cc758a3c5065b50" noinitrd vt.global_cursor_default=0 kern_guid=f935fdb6-1adb-4cf0-a184-1ec91e16c2dd cpuidle.governor=teo  
xorg-server 2:21.1.3-2ubuntu2.5 (For technical support please see http://www.ubuntu.com/support) 
Current version of pixman: 0.40.0
        Before reporting problems, check http://wiki.x.org
        to make sure that you have the latest version.
Markers: (--) probed, (**) from config file, (==) default setting,
        (++) from command line, (!!) notice, (II) informational,
        (WW) warning, (EE) error, (NI) not implemented, (??) unknown.
(++) Log file: "/tmp/Xorg.crouton.1.log", Time: Thu Dec 15 13:31:50 2022
(==) Using system config directory "/usr/share/X11/xorg.conf.d"
(EE) 
Fatal server error:
(EE) AddScreen/ScreenInit failed for driver 0
(EE) 
(EE) 
Please consult the The X.Org Foundation support 
         at http://wiki.x.org
 for help. 
(EE) Please also check the log file at "/tmp/Xorg.crouton.1.log" for additional information.
(EE) 
(EE) Server terminated with error (1). Closing log file.
/usr/bin/xinit: giving up
/usr/bin/xinit: unable to connect to X server: Connection refused
/usr/bin/xinit: server error
Unmounting /mnt/stateful_partition/crouton/chroots/jammy...

#### If known, describe the steps to reproduce the issue:
1. sudo crouton -r jammy -t xfce-desktop (xfce/gnome/gnome-desktop)
2. sudo startxfce4/startgnome 

focal  and buster are fine.

**Comment by dnastase** [comment]
Yes, I've hit that as well. Also not working for kinetic. Impish (my old install that I was trying to upgrade) works correctly though.

**Comment by hungshe** [comment]
> Yes, I've hit that as well. Also not working for `kinetic`. `impish` (my old install that I was trying to upgrade) works correctly though. Grepping the errors (EE) in the log file shows:
> 
> ```
> grep EE /tmp/Xorg.crouton.1.log
> [  1722.008] Current Operating System: Linux localhost 5.4.217-19684-g8dacb33c82e1 #1 SMP PREEMPT Wed Dec 7 23:37:59 PST 2022 x86_64
>         (WW) warning, (EE) error, (NI) not implemented, (??) unknown.
> [  1722.011] (EE) systemd-logind: failed to get session: Launch helper exited with unknown return code 1
> [  1722.013] (EE) Failed to load module "fbdev" (module does not exist, 0)
> [  1722.013] (EE) Failed to load module "vesa" (module does not exist, 0)
> [  1722.013] (EE) Failed to load module "fbdev" (module does not exist, 0)
> [  1722.013] (EE) Failed to load module "vesa" (module does not exist, 0)
> [  1722.038] (EE) modeset(0): drmSetMaster failed: Permission denied
> [  1722.038] (EE) 
> [  1722.038] (EE) AddScreen/ScreenInit failed for driver 0
> [  1722.038] (EE) 
> [  1722.038] (EE) 
> [  1722.038] (EE) Please also check the log file at "/tmp/Xorg.crouton.1.log" for additional information.
> [  1722.038] (EE) 
> [  1722.038] (EE) Server terminated with error (1). Closing log file.
> ```
> 
> _However_: I have to say that I usually use `xiwi` and I didn't pay attention to how I installed `jammy` and it defaulted to `xorg` but that's not what I wanted. Redoing the installation with `xiwi` target fixed the xfce4 for me !!

Can you share your configuration? I stiil face issue by using cli-extra,core,xfce-desktop,xiwi environments

**Comment by penguingbot7** [comment]
the new kernel does not support DRM modes is seems the server fails

**Comment by dnschneid** [comment]
@dnastase what's your output of `sudo edit-chroot -all`?

**Comment by dnastase** [comment]
After installing the `xiwi` target: 

```
name: jammy
encrypted: no
Entering /mnt/stateful_partition/crouton/chroots/jammy...
/usr/local/bin/crash_reporter_wrapper: 119: exec: /var/run/crw/crw: Permission denied
WARNING: Unable to register core dump handler.
crouton: version 1-20221126232257~master:ae594340
release: jammy
architecture: amd64
xmethod: xiwi
targets: xfce,xiwi,core,cli-extra
host: version 15183.69.0 (Official Build) stable-channel volteer 
kernel: Linux localhost 5.4.217-19684-g8dacb33c82e1 #1 SMP PREEMPT Wed Dec 7 23:37:59 PST 2022 x86_64 x86_64 x86_64 GNU/Linux
freon: yes
```

**Comment by dnastase** [comment]
Before installing the `xiwi` target (ie: w/ the default `xorg`):

```
name: kinetic
encrypted: no
Entering /mnt/stateful_partition/crouton/chroots/kinetic...
/usr/local/bin/crash_reporter_wrapper: 119: exec: /var/run/crw/crw: Permission denied
WARNING: Unable to register core dump handler.
crouton: version 1-20221126232257~master:ae594340
release: kinetic
architecture: amd64
xmethod: xorg
targets: xfce
host: version 15183.69.0 (Official Build) stable-channel volteer 
kernel: Linux localhost 5.4.217-19684-g8dacb33c82e1 #1 SMP PREEMPT Wed Dec 7 23:37:59 PST 2022 x86_64 x86_64 x86_64 GNU/Linux
freon: yes
```
(it's a different version of ubuntu but behavior is the same)

**Comment by RyelandSchool** [comment]
Sorry this info is going to be a bit dry, but I currently don't have my chromebook on me so here it goes anyway. I have been trying to get Ubuntu Jammy Jellyfish working using crouton and am running into issues as well.

When trying to get Jammy to run with gnome, in the `Xorg.crouton.1.log` file I got the same error as @NarkyAbyss: "drmSetMaster failed permission denied". After a little bit of inspection it appeared that the file /sys/kernel//debug/dri/0/clients had "chrome" labeled as the master, so I assume that is why the permission was denied. Hope this info helps debugging, cheers.
